### ISSUE TITLE
Extra icons plugin fails to distinguish between Angular and NestJS icons

### ISSUE THREAD

**Issue by tahesse** [jonathanlermitage/intellij-extra-icons-plugin#81]
## Issues

Please describe the issue:

* the plugin version
  * 1.69.3.203
* the OS and IDE version. Open your IDE and go to <kbd>Help</kbd>, <kbd>About</kbd>, then <kbd>Copy</kbd>

```
IntelliJ IDEA 2021.2.3 (Ultimate Edition)
Build #IU-212.5457.46, built on October 12, 2021
Licensed to <REDACTED>
Subscription is active until <REDACTED>
Runtime version: 11.0.11+9-b1504.8 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.10.93-1-manjaro
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 12
Registry: documentation.show.toolbar=true, ide.balloon.shadow.size=0
Non-Bundled Plugins: <REDACTED>
Current Desktop: GNOME
```

* the stacktrace, if any in the event log (in the bottom-right corner of your IDE)
  * None 
* any useful information
  * None

**Comment by jonathanlermitage** [comment]
@tahesse Can you please provide a screenshot, or file names with expected and actual icons? Thx!

**Comment by tahesse** [comment]
Hey @jonathanlermitage, sorry for the delay on my side!

My angular project looks like this:
![image](https://user-images.githubusercontent.com/13452321/161098488-17fb3abd-2470-44f8-b6c5-748759130065.png)

and my options look like this:
![image](https://user-images.githubusercontent.com/13452321/161098716-c359869b-e43f-4974-a336-dd6d198243ba.png)

**Comment by tahesse** [comment]
@jonathanlermitage Any updates on this issue?

**Comment by jonathanlermitage** [comment]
@tahesse I think this is expected behavior.  
If you work with Angular and mostly NestJS projects, you can probably proceed this way:
- update the Extra Icons settings at IDE level and disable Angular icons (you already do this)
- then, on Angular projects, enable the Extra Icons settings at Project level, enable Angular icons, and don't forget to check "Override IDE settings"
![config](https://user-images.githubusercontent.com/857535/164747005-8a949155-07c5-4c9a-9b4a-9555934481cb.png)

Otherwise, I see you are running IJ Ultimate. That means you have the Angular plugin. You will see that Extra Icons offers two sets of Angular icons: 
- `Angular` -> enabled for any IDE and all projects
- `Angular (in Angular 2+...)` -> these icons are enabled only if the (paid) Angular plugin is installed, and if this is an Angular project
I think you could disable `Angular` and keep `Angular (in Angular 2+...)`. You can configure this in at IDE level.

**Comment by jonathanlermitage** [comment]
woops, `Angular (in Angular 2+...)` icons has lowest priority. I will fix that.

**Comment by tahesse** [comment]
@jonathanlermitage Can't you check if there is an angular.json in the project? And in case there is one, show the angular icons and if not, show the nestjs ones?

**Comment by jonathanlermitage** [comment]
@tahesse Sure!  
I have a limited experience with Angular -> if my understanding is correct, modern Angular projects always have a `angular.json` file? I thought Angular was declared in a `packages.json` file only, but last time I worked with Angular it was Angular 1.2 or 1.3, something like that 😅, I'm totally lost now.

**Comment by tahesse** [comment]
> @tahesse Sure!  
> I have a limited experience with Angular -> if my understanding is correct, modern Angular projects always have a `angular.json` file? I thought Angular was declared in a `packages.json` file only, but last time I worked with Angular it was Angular 1.2 or 1.3, something like that 😅, I'm totally lost now.

Things changed. Angular consistently has an angular.json in the root folder (to the best of my knowledge). Angular version is at 13 now.

**Comment by jonathanlermitage** [comment]
Fixed. Published 2022.1.3 release.
